public/app/plugins/datasource/elasticsearch/components/QueryEditor/ElasticsearchQueryContext.tsx

```diff
@@ -31,9 +31,11 @@ export const ElasticsearchProvider = ({
   range,
 }: PropsWithChildren<Props>) => {
   const onStateChange = useCallback(
-    (query: ElasticsearchQuery) => {
+    (query: ElasticsearchQuery, prevQuery: ElasticsearchQuery) => {
       onChange(query);
-      onRunQuery();
+      if (query.query === prevQuery.query || prevQuery.query === undefined) {
```

Comment:
Anecdotically, running the query when it was previously undefined was required to make a test case pass, which was the correct implementation. This highlights again how important is to have quality unit tests, not only to prevent regressions, but to allow for refactors that don't break functionality.